tough/src/editor/mod.rs

```diff
@@ -287,42 +389,217 @@ impl RepositoryEditor {
         self
     }
 
-    // =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+    /// Takes the current Targets from `targets_editor` and inserts the role to its proper place in `signed_targets`
+    /// Sets `targets_editor` to None
+    /// Must be called before `change_delegated_targets()`
+    pub fn sign_targets_editor(&mut self, keys: &[Box<dyn KeySource>]) -> Result<&mut Self> {
+        if let Some(targets_editor) = self.targets_editor.as_mut() {
+            let (name, targets) = targets_editor.create_signed(keys)?.targets();
+            if name == "targets" {
+                self.signed_targets = Some(targets);
+            } else {
+                self.signed_targets
+                    .as_mut()
+                    .context(error::NoTargets)?
+                    .signed
+                    .delegated_role_mut(&name)
+                    .context(error::DelegateMissing { name })?
+                    .targets = Some(targets);
+            }
+        }
+        self.targets_editor = None;
+        Ok(self)
+    }
 
-    /// Build the `Targets` struct
-    fn build_targets(&self) -> Result<Targets> {
-        let version = self.targets_version.context(error::Missing {
-            field: "targets version",
-        })?;
-        let expires = self.targets_expires.context(error::Missing {
-            field: "targets expiration",
-        })?;
+    /// Changes the targets refered to in `targets_editor` to role
```

Comment:
Let's add a little bit more to this docstring in case a user didn't read the docstring on the RepositoryEditor.  Let's further explain what's going to happen here and in what situations would a user want to use this.